### ISSUE TITLE
[FR] horizontal gallery improvements : display uncropped images  ?

### ISSUE THREAD

**Issue by cannibalox** [Ruben2776/PicView#53]
**Is your feature request related to a problem? Please describe.**
some suggestions for the horizontal gallery view:
- add an option to display the horizontal gallery thumbnails with fixed height but adaptative width so that they are not cropped to square ratio, as in the example below:
![image](https://user-images.githubusercontent.com/4605693/140555963-d548ac76-dbea-451d-9a4f-547b3d4a60fd.png)

- additionnal setting for custom thumbnail height  (slider ?) 
- additionnal adjustments for padding/margin between the images

- also the thumbnail generation takes a very long time compared to other viewers in the current version (test folder containing 100 pics of 4860x3280 on a network folder, current picview takes about 600-700ms to display a thumbnail, see the gif below. xnview is about x10 times faster on first navigation without using thumbnail cache/indexing)
![R1700-LAST10_20211105_1204_jaWYZ5zEW2](https://user-images.githubusercontent.com/4605693/140557185-f6a4e6e2-3685-4d00-ba7c-ce7d623f631f.gif)

**Comment by Ruben2776** [comment]
For thumbnail generation, I have two methods. One that is multi threaded/asynchronous and more resource intensive, but it only fires if the amount of images is at or above 250. What you are seeing is a background single threaded use. 
In hindsight, I should've done more testing with it and maybe put the number at 50 or perhaps only use the multi threaded solution.
The multi threaded use seems to be acceptably fast in my folder at about 2,200 pics, but then again I have an overclocked gaming PC. 

I have uploaded a new test version to Sourceforge, where the condition is set to 40 images instead of 250. Tell me if that works better for you.
https://sourceforge.net/projects/picview/files/PicView-v1.5.6_win-x64-portable-.NET6-Req%20Gallery%20Test.zip/download


For the size and padding, let me think about it and get back to you on a later date.

**Comment by Ruben2776** [comment]
Alright, progress report.

With release 1.6 I have improved the thumbnail gallery loading time. It should be a lot faster now,

For resizing thumbnail dimensions, that will be a bigger project. It is something I want to accomplish.
When you enter the gallery, it scrolls to the selected item, which is somewhat easy to calculate since they all have the same dimension, so the program knows where to scroll immediately. When they have various sizes, I do not know how to calculate that.
I'm not the sharpest with numbers, so if anyone could give me a helping hand that would be great. 

I'll start working on it after I've created my image resizing window.
